Source/Views/TableViews/SPCopyTable.m

```diff
@@ -1361,217 +1361,220 @@ - (IBAction)executeBundleItemForDataTable:(id)sender
-				action = SPBundleOutputActionReplaceSelection;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionReplaceContent) {
-				action = SPBundleOutputActionReplaceContent;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionInsertAsText) {
-				action = SPBundleOutputActionInsertAsText;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionInsertAsSnippet) {
-				action = SPBundleOutputActionInsertAsSnippet;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionShowAsHTML) {
-				action = SPBundleOutputActionShowAsHTML;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionShowAsTextTooltip) {
-				action = SPBundleOutputActionShowAsTextTooltip;
-				err = nil;
-			}
-			else if([err code] == SPBundleRedirectActionShowAsHTMLTooltip) {
-				action = SPBundleOutputActionShowAsHTMLTooltip;
-				err = nil;
-			}
-		}
+    NSString *action = SPBundleOutputActionNone;
+    if([cmdData objectForKey:SPBundleFileOutputActionKey] && [(NSString *)[cmdData objectForKey:SPBundleFileOutputActionKey] length])
+        action = [[cmdData objectForKey:SPBundleFileOutputActionKey] lowercaseString];
 
-		if(err == nil && output) {
-			if(![action isEqualToString:SPBundleOutputActionNone]) {
-				NSPoint pos = [NSEvent mouseLocation];
-				pos.y -= 16;
+    // recompute action and err
+    if(err != nil) {
+        action = [SPBundleCommandRunner computeActionWithError: &err];
+    }
 
-				if([action isEqualToString:SPBundleOutputActionShowAsTextTooltip] && [output length] > 0) {
-					[SPTooltip showWithObject:output atLocation:pos];
-				}
+    if ([err code] != 9) { // Suppress an error message if command was killed
+        NSString *errorMessage  = [err localizedDescription];
+        [NSAlert createWarningAlertWithTitle:NSLocalizedString(@"BASH Error", @"bash error") message:[NSString stringWithFormat:@"%@ “%@”:\n%@", NSLocalizedString(@"Error for", @"error for message"), [cmdData objectForKey:@"name"], errorMessage] callback:nil];
+    }
+    
+    if (!output || [action isEqualToString:SPBundleOutputActionNone]) {
+        return;
+    }
+    
+    NSPoint pos = [NSEvent mouseLocation];
+    pos.y -= 16;
 
-				else if([action isEqualToString:SPBundleOutputActionShowAsHTMLTooltip]) {
-					[SPTooltip showWithObject:output atLocation:pos ofType:@"html"];
-				}
+    if([action isEqualToString:SPBundleOutputActionShowAsTextTooltip] && [output length] > 0) {
+        [SPTooltip showWithObject:output atLocation:pos];
+    }
+
+    else if([action isEqualToString:SPBundleOutputActionShowAsHTMLTooltip]) {
+        [SPTooltip showWithObject:output atLocation:pos ofType:@"html"];
+    }
+
+    else if([action isEqualToString:SPBundleOutputActionShowAsHTML]) {
+        BOOL correspondingWindowFound = NO;
+        for(id win in [NSApp windows]) {
+            if([[win delegate] isKindOfClass:[SPBundleHTMLOutputController class]]) {
```

Comment:
Can we combine these ifs?

Reply:
I tried, but I think that the XCode leverages the separate `if` statements to do some type inference magic. (I got complier warnings when I `&&`ed them together).

 I've added an explicit cast to make the mechanics of this clearer.